internal/sdkv2provider/resource_cloudflare_list_item.go

```diff
@@ -0,0 +1,217 @@
+	"github.com/cloudflare/terraform-provider-cloudflare/internal/consts"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareListItem() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareListItemSchema(),
+		CreateContext: resourceCloudflareListItemCreate,
+		ReadContext:   resourceCloudflareListItemRead,
+		UpdateContext: resourceCloudflareListItemUpdate,
+		DeleteContext: resourceCloudflareListItemDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCloudflareListItemImport,
+		},
+		Description: heredoc.Doc(`
+			Provides individual list items (IPs, Redirects) to be used in Edge Rules Engine
+			across all zones within the same account.
+		`),
+	}
+}
+
+func resourceCloudflareListItemCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get(consts.AccountIDSchemaKey).(string)
+	listID := d.Get("list_id").(string)
+	listItemType := listItemType(d)
+
+	list, err := client.GetList(ctx, cloudflare.AccountIdentifier(accountID), listID)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("unable to find list with id %s: %w", listID, err))
+	}
+
+	if list.Kind != listItemType {
+		return diag.FromErr(fmt.Errorf("items of type %s can not be added to lists of type %s", listItemType, list.Kind))
+	}
+
+	createListItemResponse, err := client.CreateListItem(ctx, cloudflare.AccountIdentifier(accountID), cloudflare.ListCreateItemParams{
+		ID:   listID,
+		Item: buildListItemCreateRequest(d),
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("unable to create list item on list id %s: %w", listID, err))
+	}
+
+	newestItem := mostRecentlyCreatedItem(createListItemResponse)
+	d.SetId(newestItem.ID)
+
+	return nil
+}
+
+func resourceCloudflareListItemImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	attributes := strings.SplitN(d.Id(), "/", 3)
+
+	if len(attributes) != 3 {
+		return nil, fmt.Errorf("invalid id (\"%s\") specified, should be in format \"accountID/listID/itemID\"", d.Id())
+	}
+
+	accountID, listID, itemID := attributes[0], attributes[1], attributes[2]
+	d.SetId(itemID)
+	d.Set(consts.AccountIDSchemaKey, accountID)
+	d.Set("list_id", listID)
+
+	resourceCloudflareListItemRead(ctx, d, meta)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceCloudflareListItemRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get(consts.AccountIDSchemaKey).(string)
+	listID := d.Get("list_id").(string)
+
+	listItem, err := client.GetListItem(ctx, cloudflare.AccountIdentifier(accountID), listID, d.Id())
+	if err != nil {
+		if strings.Contains(err.Error(), "could not find item") {
```

Comment:
instead of using string contains matches, let's us the `*cloudflare.NotFoundError` error directly. you can see an example of this at https://github.com/cloudflare/terraform-provider-cloudflare/blob/144b7908512e82e8415d039084d2519221d2008c/internal/sdkv2provider/resource_cloudflare_filter.go#L89-L90